### PR TITLE
[freeglut] Don't enforce x64 when the compiler is 64 bit

### PR DESCRIPTION
**Pull request by davidebeatrici**
This fixes compilation for ARM64.

### CODE REVIEW

File: ports/freeglut/no_x64_enforcement.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@@ -418,12 +418,6 @@ IF(WIN32)
+         IF(FREEGLUT_REPLACE_GLUT)
+             SET_TARGET_PROPERTIES(freeglut_static PROPERTIES OUTPUT_NAME ${LIBNAME})
+         ENDIF()
+-        # need to set machine:x64 for linker, at least for VC10, and
+-        # doesn't hurt for older compilers:
+-        # http://public.kitware.com/Bug/view.php?id=11240#c22768
+-        IF (CMAKE_CL_64)
+-            SET_TARGET_PROPERTIES(freeglut_static PROPERTIES STATIC_LIBRARY_FLAGS "/machine:x64")
+-        ENDIF()
```
Comment (JackBoosY) [inline ports/freeglut/no_x64_enforcement.patch:14]: Does this bug fixed? Please tell me your reason to remove that.

Thanks.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Please bump the port version. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).
Also please run command `vcpkg x-add-version --all` then commit the changes.

File: ports/freeglut/no_x64_enforcement.patch (hunk 1 above)
Comment (davidebeatrici) [reply ports/freeglut/no_x64_enforcement.patch:14]: Build fails because both `/machine:arm64` and `/machine:x64` are passed, my patch fixes the issue.

**Comment by davidebeatrici** [comment]
Please let me know whether I should squash the commits.

**Comment by JackBoosY** [comment]
Please run command `vcpkg x-add-version freeglut`.

**Comment by davidebeatrici** [comment]
I did.

File: ports/freeglut/CONTROL [hunk 2]
```diff
@@ -1,4 +1,5 @@
 Source: freeglut
 Version: 3.2.1-4
+Port-Version: 1
```
Comment (JackBoosY) [inline ports/freeglut/CONTROL:3]:
```suggestion
Version: 3.2.1
Port-Version: 5
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: versions/f-/freeglut.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "8657a040f529b17524200caa49ab7889ab412229",
+      "version-string": "3.2.1-4",
+      "port-version": 1
```
Comment (JackBoosY) [inline versions/f-/freeglut.json:6]:
```suggestion
      "version-string": "3.2.1",
      "port-version": 5
```

File: versions/baseline.json [hunk 4]
```diff
@@ -2038,7 +2038,7 @@
     },
     "freeglut": {
       "baseline": "3.2.1-4",
```
Comment (JackBoosY) [inline versions/baseline.json:2040]:
```suggestion
      "baseline": "3.2.1",
```

File: versions/baseline.json [hunk 5]
```diff
@@ -2038,7 +2038,7 @@
...
-      "port-version": 0
+      "port-version": 1
```
Comment (JackBoosY) [inline versions/baseline.json:2041]:
```suggestion
      "port-version": 5
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: versions/f-/freeglut.json [hunk 6]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "8657a040f529b17524200caa49ab7889ab412229",
```
Comment (JackBoosY) [inline versions/f-/freeglut.json:4]:
```suggestion
      "git-tree": "bfc1c37b44246e2a3c8bfead99af1d14a2103453",
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**